File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.py
from collections import defaultdict
class Solution:
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:
        i, j = 0, 0
        cnt = ans = 0
        # Sliding window
        while j < len(nums):
            if nums[j] % 2:
                k -= 1
                cnt = 0
            while k == 0:
                if nums[i] % 2:
                    k += 1
                cnt += 1
                i += 1
            ans += cnt
            j += 1
        return ans

File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var numberOfSubarrays = function(nums, k) {
    let i = 0, j = 0, cnt = 0, ans = 0;
    while (j<nums.length){
        if(nums[j] % 2){
            cnt = 0;
            k--;
        }
        while (k == 0){
            if (nums[i]%2){
                k++;
            }
            i++;
            cnt++;
        }
        ans += cnt;
        j++;
    }
    return ans;
    
};